GpxExport/Models/Workout+FitWriter.swift

```diff
@@ -0,0 +1,163 @@
+//
+//  Workout+FitWriter.swift
+//  GpxExport
+//
+//  Created by Mario Martelli on 23.02.19.
+//  Copyright © 2019 Mario Martelli. All rights reserved.
+//
+
+import Foundation
+import HealthKit
+import CoreLocation
+import FitDataProtocol
+import FitnessUnits
+import AntMessageProtocol
+
+extension Workout {
+    func writeFit() -> URL? {
+
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd HH.mm.ss"
+
+        let fileName = "\(formatter.string(from: startDate)) - \(activityType)"
+
+        let targetURL = URL(fileURLWithPath: NSTemporaryDirectory())
+            .appendingPathComponent(fileName)
+            .appendingPathExtension("fit")
+
+        DispatchQueue.global(qos: .background).sync {
+            let time = FitTime(date: self.startDate)
+            let serial = ValidatedBinaryInteger(value: UInt32(123), valid: true)
+
+            let file: FileHandle
+
+            do {
+                let manager = FileManager.default
+                if manager.fileExists(atPath: targetURL.path) {
+                    try manager.removeItem(atPath: targetURL.path)
+                }
+                manager.createFile(atPath: targetURL.path, contents: Data())
+                file = try FileHandle(forWritingTo: targetURL)
+            } catch let err {
```

Comment:
Btw. you can also access error without the `let`.

```
} catch {
 print(error)
}
```

I also would output `error.debugDescription` or `error.localizedDescription`.